### ISSUE TITLE
Crash (some times) at the beginning of training

### ISSUE THREAD

**Issue by mathmanu** [NVIDIA/caffe#363]
[cifar10_jacintonet11v2_2017-06-28_09-11-20_random_crash.zip](https://github.com/NVIDIA/caffe/files/1107513/cifar10_jacintonet11v2_2017-06-28_09-11-20_random_crash.zip)
Sometimes, I get a crash at the beginning of the training. This crash happens only once in a while.

Copied the last part of the log: (The full log is in the train.log file - in the attachment).

I0628 09:11:22.737768 14946 cudnn_conv_layer.cpp:833] [0] Conv Algos (F,BD,BF): 'res5a_branch2b' with space 0.02G/2 7 5 5  (limit 7.57G, req 0.02G)
I0628 09:11:22.762594 14943 internal_thread.cpp:40] Restarting 4 internal thread(s) on device 0
I0628 09:11:22.762696 14943 data_reader.cpp:52] Data Reader threads: 4, out queues: 16, depth: 32
I0628 09:11:22.762763 14949 internal_thread.cpp:40] Restarting 4 internal thread(s) on device 1
I0628 09:11:22.762763 14943 internal_thread.cpp:19] Starting 4 internal thread(s) on device 0
I0628 09:11:22.763093 14949 data_reader.cpp:52] Data Reader threads: 4, out queues: 16, depth: 32
I0628 09:11:22.763219 14949 internal_thread.cpp:19] Starting 4 internal thread(s) on device 1
I0628 09:11:22.763523 14973 db_lmdb.cpp:35] Opened lmdb ./data/cifar10_train_lmdb
I0628 09:11:22.763557 14943 data_layer.cpp:188] ReshapePrefetch 32, 3, 32, 32
I0628 09:11:22.764204 14975 db_lmdb.cpp:35] Opened lmdb ./data/cifar10_train_lmdb
I0628 09:11:22.764447 14943 data_layer.cpp:206] Output data size: 32, 3, 32, 32
I0628 09:11:22.764750 14976 db_lmdb.cpp:35] Opened lmdb ./data/cifar10_train_lmdb
I0628 09:11:22.765700 14978 db_lmdb.cpp:35] Opened lmdb ./data/cifar10_train_lmdb
I0628 09:11:22.766468 14982 db_lmdb.cpp:35] Opened lmdb ./data/cifar10_train_lmdb
I0628 09:11:22.766496 14949 data_layer.cpp:188] ReshapePrefetch 32, 3, 32, 32
I0628 09:11:22.767531 14984 db_lmdb.cpp:35] Opened lmdb ./data/cifar10_train_lmdb
I0628 09:11:22.768359 14985 db_lmdb.cpp:35] Opened lmdb ./data/cifar10_train_lmdb
I0628 09:11:22.768815 14949 data_layer.cpp:206] Output data size: 32, 3, 32, 32
I0628 09:11:22.782820 14983 db_lmdb.cpp:35] Opened lmdb ./data/cifar10_train_lmdb
I0628 09:11:22.786026 14943 data_layer.cpp:110] [0] Parser threads: 4 (auto)
I0628 09:11:22.786041 14943 data_layer.cpp:112] [0] Transformer threads: 4 (auto)
I0628 09:11:22.786106 14946 solver.cpp:354] Iteration 0 (0.15107 s), loss = 2.41522
I0628 09:11:22.786134 14946 solver.cpp:371]     Train net output #0: loss = 2.41522 (* 1 = 2.41522 loss)
I0628 09:11:22.786159 14946 sgd_solver.cpp:137] Iteration 0, lr = 0.1, m = 0.9
I0628 09:11:22.786535 14949 data_layer.cpp:110] [1] Parser threads: 4 (auto)
I0628 09:11:22.786545 14949 data_layer.cpp:112] [1] Transformer threads: 4 (auto)
I0628 09:11:22.786775 14972 internal_thread.cpp:40] Restarting 4 internal thread(s) on device 0
F0628 09:11:22.786790 14972 internal_thread.cpp:46] Check failed: 1 == threads_.size() (1 vs. 4) 
*** Check failure stack trace: ***
    @     0x7f6be49e5daa  (unknown)
    @     0x7f6be49e5ce4  (unknown)
    @     0x7f6be49e56e6  (unknown)
    @     0x7f6be49e8687  (unknown)
    @     0x7f6be532c6ff  caffe::InternalThread::RestartAllThreads()
    @     0x7f6be542d0a0  caffe::DataLayer<>::InitializePrefetch()
    @     0x7f6be550f31b  caffe::BasePrefetchingDataLayer<>::InternalThreadEntryN()
    @     0x7f6be532bebc  caffe::InternalThread::entry()
    @     0x7f6be532cb44  boost::detail::thread_data<>::run()
    @     0x7f6bda132e7a  (unknown)
    @     0x7f6be2bd5184  start_thread
    @     0x7f6be3103bed  (unknown)
    @              (nil)  (unknown)
Aborted (core dumped)

**Comment by drnikolaev** [comment]
Hi @mathmanu thanks! Yeah, we noticed it too and already fixed. New release is coming soon.

**Comment by coolper** [comment]
@drnikolaev can you share some detail about how to fix it?
I need it in a hurry.
(I get this crash every time in my environment, it blocks me to contine.)
thx.

**Comment by drnikolaev** [comment]
Got back from vacation, now polishing this...

**Comment by drnikolaev** [comment]
Fixed in 0.16.3. Thanks for reporting this.
